src/EventStore.LogV3/Raw.cs

```diff
@@ -33,8 +31,9 @@ public struct RecordHeader {
 			}
 
 			public DateTime TimeStamp {
-				get => new DateTime(_ticks);
-				set => _ticks = value.Ticks;
+				// little-endian. blank out the 2 LEAST significant bytes
+				get => new(_ticks & SixMostSignificantBytes);
+				set => _ticks |= value.Ticks & SixMostSignificantBytes;
```

Comment:
should this be: `_ticks = value.Ticks & SixMostSignificantBytes` ?

Reply:
we need the |= so that it doesn't overwrite the two least significant bytes

Reply:
@timothycoleman  ah ok, got it. the issue I was trying to convey (before I understood the purpose of the OR) is that the 6 most significant bytes from the existing value of ticks are being ORed into the result. so setting _ticks twice could result in an invalid value.

this would prevent it:
`_ticks = (_ticks & ~SixMostSignificantBytes) | (value.Ticks & SixMostSignificantBytes)`

Reply:
🤦🤦🤦 thanks

Reply:
fixed, nice save that could have been a pain!

Reply:
Shouldn't this be covered by a test?

Reply:
@pvanbuijtene it is now, the reason it wasn't before is because is requires two assignments to show the problem and the original test was just setting then checking. in practice the records are immutable after they have been created so calling the setter more than once shouldn't happen, but still its nice to be able to of course